server/channels/api4/user_test.go

```diff
@@ -7917,3 +7919,559 @@ func TestLoginWithDesktopToken(t *testing.T) {
+		// We should check that the returned stats make sense in this context
+		require.Equal(t, int64(0), stats.TotalUsersCount, "Expected 0 users for an invalid team ID")
+	})
+
+    t.Run("Get filtered users stats with roles", func(t *testing.T) {
+        options := &model.UserCountOptions{
+            TeamId: th.BasicTeam.Id,
+            IncludeDeleted: false,
+            IncludeBotAccounts: false,
+            IncludeRemoteUsers: true,
+            Roles: []string{model.SystemUserRoleId},
+        }
+        stats, resp, err := th.SystemAdminClient.GetFilteredUsersStats(context.Background(), options)
+        require.NoError(t, err)
+        CheckOKStatus(t, resp)
+        require.NotNil(t, stats)
+        require.True(t, stats.TotalUsersCount > 0)
+    })
+
+    t.Run("Get filtered users stats with team roles", func(t *testing.T) {
+        options := &model.UserCountOptions{
+            TeamId: th.BasicTeam.Id,
+            IncludeDeleted: false,
+            IncludeBotAccounts: false,
+            IncludeRemoteUsers: true,
+            TeamRoles: []string{model.TeamUserRoleId},
+        }
+        stats, resp, err := th.SystemAdminClient.GetFilteredUsersStats(context.Background(), options)
+        require.NoError(t, err)
+        CheckOKStatus(t, resp)
+        require.NotNil(t, stats)
+        require.True(t, stats.TotalUsersCount > 0)
+    })
+}
+
+func TestGetDefaultProfileImage(t *testing.T) {
+    th := Setup(t).InitBasic()
+    defer th.TearDown()
+
+    t.Run("Get default profile image for existing user", func(t *testing.T) {
+        user := th.BasicUser
+
+        img, resp, err := th.Client.GetDefaultProfileImage(context.Background(), user.Id)
+        require.NoError(t, err)
+        require.NotNil(t, img)
+        require.Equal(t, http.StatusOK, resp.StatusCode)
+        
+        // Check if the image is a valid PNG
+        _, err = png.Decode(bytes.NewReader(img))
+        require.NoError(t, err, "Image should be a valid PNG")
+    })
+
+    t.Run("Get default profile image for non-existent user", func(t *testing.T) {
+        nonExistentUserId := model.NewId()
+
+        _, resp, err := th.Client.GetDefaultProfileImage(context.Background(), nonExistentUserId)
+        require.Error(t, err)
+        CheckNotFoundStatus(t, resp)
+    })
+
+    t.Run("Get default profile image without proper permissions", func(t *testing.T) {
+        user := th.CreateUser()
+
+        th.Client.Logout(context.Background())
+        _, resp, err := th.Client.GetDefaultProfileImage(context.Background(), user.Id)
+        require.Error(t, err)
+        CheckUnauthorizedStatus(t, resp)
+    })
+
+    t.Run("Get default profile image as system admin", func(t *testing.T) {
+        user := th.CreateUser()
+
+        img, resp, err := th.SystemAdminClient.GetDefaultProfileImage(context.Background(), user.Id)
+        require.NoError(t, err)
+        require.NotNil(t, img)
+        require.Equal(t, http.StatusOK, resp.StatusCode)
```

Comment:
Maybe worth decoding the image here as well? And maybe assert expected image is returned like a few tests earlier?